src/memory_targets/memory_target_numa.c

```diff
@@ -0,0 +1,131 @@
+/*
+ *
+ * Copyright (C) 2023 Intel Corporation
+ *
+ * Under the Apache License v2.0 with LLVM Exceptions. See LICENSE.TXT.
+ * SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+ *
+ */
+
+#include <assert.h>
+#include <stdlib.h>
+
+#include "../memory_pool_internal.h"
+#include "memory_target_numa.h"
+#include <umf/pools/pool_disjoint.h>
+#include <umf/providers/provider_os_memory.h>
+
+struct numa_memory_target_t {
+    size_t id;
+};
+
+static enum umf_result_t numa_initialize(void *params, void **memTarget) {
+    if (params == NULL || memTarget == NULL) {
+        return UMF_RESULT_ERROR_INVALID_ARGUMENT;
+    }
+
+    struct umf_numa_memory_target_config_t *config =
+        (struct umf_numa_memory_target_config_t *)params;
+
+    struct numa_memory_target_t *numaTarget =
+        malloc(sizeof(struct numa_memory_target_t));
+    if (!numaTarget) {
+        return UMF_RESULT_ERROR_OUT_OF_HOST_MEMORY;
+    }
+
+    numaTarget->id = config->id;
+    *memTarget = numaTarget;
+    return UMF_RESULT_SUCCESS;
+}
+
+static void numa_finalize(void *memTarget) { free(memTarget); }
+
+static const umf_os_memory_provider_params_t
+    UMF_OS_MEMORY_PROVIDER_PARAMS_DEFAULT = {
+        // Visibility & protection
+        .protection = UMF_PROTECTION_READ | UMF_PROTECTION_WRITE,
+        .visibility = UMF_VISIBILITY_PRIVATE,
+
+        // NUMA config
+        .nodemask = NULL,
+        .maxnode = 0, // TODO: numa_max_node/GetNumaHighestNodeNumber
+        .numa_mode = UMF_NUMA_MODE_DEFAULT,
+        .numa_flags = UMF_NUMA_FLAGS_STRICT, // TODO: determine default behavior
+
+        // Logging
+        .traces = 0, // TODO: parse env variable for log level?
+};
+
+static unsigned long
+numa_targets_get_nodemask(struct numa_memory_target_t **targets,
```

Comment:
I think we should return here nodemask_t from libnuma. Please note that this mask could exceed unsigned long capacity. Also as libnuma already implements some functions that handle these types of mask I think we should use it, so please consider calling numa_bitmask_setbit() here.

Reply:
Done.

Reply:
But tbh, this will need to be changed to something cross-platform anyway. We might use hwloc instead of numa to make it work on windows as well (same is true for os provider).